File: machine-translation/ibm-watson-language-translator/IbmWatsonLanguageTranslatorMachineTranslation.java
// "input" is the string containing the input document.
// To translate from English to German, set "source" to "en" and "target" to "de".

IamOptions iamOptions = new IamOptions.Builder()
    .apiKey(apiKey)
    .build();

LanguageTranslator translator = new LanguageTranslator("2018-03-16", iamOptions);
// It may be required to override the default endpoint.
translator.setEndPoint(endPoint);

TranslateOptions options = new TranslateOptions.Builder()
    .addText(input)
    .modelId(String.format("%s-%s", source, target))
    .build();

TranslationResult result = translator
    .translate(options)
    .execute();

Translation translation = result.getTranslations().get(0);

String translatedText = translation.getTranslationOutput();

File: machine-translation/ibm-watson-language-translator/IbmWatsonLanguageTranslatorMachineTranslation.kt
// "input" is a String containing the input document.
// To translate from English to German, set "source" to "en" and "target" to "de".

val iamOptions = IamOptions.Builder()
    .apiKey(apiKey)
    .build()

val translator = LanguageTranslator("2018-03-16", iamOptions)
translator.endPoint = endPoint

val options = TranslateOptions.Builder()
    .addText(input)
    .modelId("$source-$target")
    .build()

val result = translator
    .translate(options)
    .execute()

val translation = result.translations.single()

val translatedText = translation.translationOutput
